app/src/main/java/com/rajatgoyal/puzzle15/ui/MainActivity.java

```diff
@@ -228,9 +126,16 @@ public void setLatestHighScore(HighScore latestHighScore) {
     }
 
     public void init() {
```

Comment:
Move the `init` method in BaseActivity, so that all activity inheriting `BaseActivity` has to implement this init method